evo_researcher/functions/cache.py

```diff
@@ -0,0 +1,16 @@
+import os
+from typing import TypeVar, Callable, Any, cast
+from joblib import Memory
+from functools import cache
```

Comment:
> Is this because you might have multiple evo agents in a single benchmark run?

Yeah, exactly. I had multiple evo agents with different settings (but still calling Tavily)

But in fact I didn't meausre how long it takes to load it from files, we can drop the in-memory one if you want.

Reply:
I made a toy example to test, and the in-memory load was ~100x faster, but the load from file was still very fast. Happy for both, but would get good to add a comment about the in-memory one

Reply:
There is this comment:

```
Wraps a function with both file cache (for persistent cache) and in-memory cache (for speed).
```

I'm happy to add any additional comment; I'm just not sure what you have in mind.

Reply:
Sorry, you're right, I think that's sufficient, as it's a generic function, not specific to benchmarking :)